deploy/containers/credhub_setup/auth.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+// This file contains code to deal with UAA / authentication.
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+func makeHTTPClientWithCA(serverName string, caCert []byte) (*http.Client, error) {
+	certPool, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, fmt.Errorf("could not get system cert pool: %w", err)
+	}
+	ok := certPool.AppendCertsFromPEM(caCert)
+	if !ok {
+		fmt.Printf("Warning: failed to add CA certs for %s\n", serverName)
```

Comment:
I would expect it to be an error. It's ok to ignore this particular error if the caller judges to be the correct behaviour. After all, this is the main behaviour of this function.